### ISSUE TITLE
State composition, why default state is not inherited?

### ISSUE THREAD

**Issue by gremo** [ngxs/store#574]
Maybe I misunderstood the the state composition feature. I have a `GenericListingStateModel` which provides `defaults` (model omitted for brevity):

```typescript
@State<GenericListingStateModel<T>>({
  name: 'genericList',
  defaults: {
    entities: {},
    currentPage: 1,
    currentLimit: 10,
    currentPageResults: [],
  }
})
export class GenericListingState<T> { }
```

Then I use the state composition to inherit the state model (and the defualts):

```typescript
@State<GenericListingStateModel<Customer>>({ name: 'customers' })
export class CustomerState extends GenericListingState<Customer> { }
```
Why the state of `CustomerState` **is empty and doesn't inherit all the defaults** defined in the `GenericListingState`?

**Comment by matulkum** [comment]
I think because you overwrote the @State decorator and if I am not mistaken there is no inheritance in decorators, therefore it would not be possible to inherit the _defaults_ .

**Comment by gremo** [comment]
Well, without overriding `@State` I can't assigna name to my `CustomerState `. So... what's the point of state composition as explained [here](https://ngxs.gitbook.io/ngxs/advanced/composition)?

**Comment by matulkum** [comment]
you still can, you just have to apply the whole decorator again. So just pass in your defaults into the decorator of your inheriting class again.

Again: I did not use inheritance in that case myself. Someone please correct me if I am stating something false ;)

**Comment by gremo** [comment]
@matulkum duplcation of code. Defaults must be applied both to the parent class and to the child class (correct me if I'm wrong).

**Comment by matulkum** [comment]
yeah.. kind of. But it is not so bad. I guess you could do it like this:
```js
const defaults = {
  entities: {},
  currentPage: 1,
  currentLimit: 10,
  currentPageResults: [],
}

@State<GenericListingStateModel<T>>({
  name: 'genericList',
  defaults,
})
export class GenericListingState<T> { }

@State<GenericListingStateModel<Customer>>({
  name: 'customers',
  defaults,
})
export class CustomerState extends GenericListingState<Customer> { }
```

**Comment by juliusstoerrle** [comment]
@gremo you mentioned a state composition feature, could you (maybe by providing a link to the docs), show which feature you ment?

**Comment by gremo** [comment]
@juliusstoerrle https://ngxs.gitbook.io/ngxs/advanced/composition

**Comment by juliusstoerrle** [comment]
Thanks, I never saw that feature until today..
Could you create a simple repro in https://stackblitz.com? Then we could try to figure this out and either adjust the docs or implement inheritance of the defaults, if possible.

**Comment by markwhitfeld** [comment]
Technically, this should only be closed once 3.4.0 is released.
